File: python/mix/D02_matrix_multiply.py
def matmul(A, B):
    m, n, p = len(A), len(A[0]), len(B[0])
    assert len(B) == n
    C = [[0]*p for _ in range(m)]
    for i in range(m):
        for k in range(n):
            aik = A[i][k]
            for j in range(p):
                C[i][j] += aik * B[k][j]
    return C

File: javascript/mix/D02MatrixMultiply.js
function matmul(A, B) {
    const m = A.length;
    const n = A[0].length;
    const p = B[0].length;
    console.assert(B.length === n, "Matrix dimension mismatch");

    const C = Array.from({ length: m }, () => Array(p).fill(0));

    for (let i = 0; i < m; i++) {
        for (let k = 0; k < n; k++) {
            const aik = A[i][k];
            for (let j = 0; j < p; j++) {
                C[i][j] += aik * B[k][j];
            }
        }
    }
    return C;
}

module.exports = { matmul };
